XForm/XForm/Functions/String/Replace.cs

```diff
@@ -0,0 +1,142 @@
+    {
+        private IDataBatchColumn Column { get; }
+        private String8 FindText { get; }
+        private String8 ReplaceText { get; }
+        private bool ExactMatch { get; }
+
+        private ReplaceColumn(IDataBatchColumn column, String8 findText, String8 replaceText, bool exactMatch)
+        {
+            Column = column;
+            FindText = findText;
+            ReplaceText = replaceText;
+            ExactMatch = exactMatch;
+        }
+
+        public static IDataBatchColumn Build(IDataBatchEnumerator source, IDataBatchColumn inputColumn, string findText, string replaceText, bool exactMatch)
+        {
+            // Replace only works on String8 columns
+            if (inputColumn.ColumnDetails.Type != typeof(String8))
+            {
+                return inputColumn;
+            }
+
+            String8 find = findText == null ? String8.Empty : String8.Convert(findText, new byte[String8.GetLength(findText)]);
+            String8 replace = replaceText == null ? String8.Empty : String8.Convert(replaceText, new byte[String8.GetLength(replaceText)]);
+
+            if (inputColumn is Constant)
+            {
+                // If the inner value is a constant, convert once and store the new constant
+                return new Constant(source, ReplaceColumn.Replace((String8)((Constant)inputColumn).Value, find, replace, exactMatch, new String8Block()), typeof(String8));
+            }
+            else
+            {
+                return new ReplaceColumn(inputColumn, find, replace, exactMatch);
+            }
+        }
+
+        public ColumnDetails ColumnDetails
+        {
+            get
+            {
+                return Column.ColumnDetails;
+            }
+        }
+
+        public Func<DataBatch> Getter()
+        {
+            String8[] transformedArray = null;
+            String8Block block = new String8Block();
+            return () => ReplaceBatch(Column.Getter()(), block, ref transformedArray);
+        }
+
+        private DataBatch ReplaceBatch(DataBatch batch, String8Block block, ref String8[] transformedArray)
+        {
+            block.Clear();
+            Allocator.AllocateToSize(ref transformedArray, batch.Count);
+
+            String8[] sourceArray = (String8[])batch.Array;
+            for (int i = 0; i < batch.Count; ++i)
+            {
+                String8 originalValue = sourceArray[batch.Index(i)];
+                transformedArray[i] = ReplaceColumn.Replace(originalValue, FindText, ReplaceText, ExactMatch, block);
+            }
+
+            return DataBatch.All(transformedArray, batch.Count);
+        }
+
+        public static String8 Replace(String8 originalValue, String8 findText, String8 replaceText, bool exactMatch, String8Block block)
+        {
+            String8 newValue = String8.Empty;
+            int startIndex = -1;
+            int currentIndex = 0;
+
+            if (exactMatch)
+            {
+                if (originalValue == findText)
```

Comment:
Does this turn into String8.Equals or object.Equals? Maybe change to String8.Equals to be sure we're not paying a boxing tax.